source/grid/grid_generator.cc

```diff
@@ -4991,6 +4991,281 @@ namespace GridGenerator
 
 
 
+  template <int dim>
+  void
+  non_standard_orientation_mesh(Triangulation<dim> &tria,
+                                const bool          face_orientation,
+                                const bool          face_flip,
+                                const bool          face_rotation,
+                                const bool          manipulate_left_cube)
+  {
+    if (dim < 3)
+      AssertThrow(false, ExcMessage("This mesh is only implemented in 3D."));
+
+    std::vector<Point<dim>>    vertices;
+    const unsigned int         n_cells = 2;
+    std::vector<CellData<dim>> cells(n_cells);
+
+    // Corner points of the cube [0,1]^3
+    static const std::array<Point<dim>, 12> hexahedron = {{{0, 0, 0},   // 0
+                                                           {1, 0, 0},   // 1
+                                                           {0, 1, 0},   // 2
+                                                           {1, 1, 0},   // 3
+                                                           {0, 0, 1},   // 4
+                                                           {1, 0, 1},   // 5
+                                                           {0, 1, 1},   // 6
+                                                           {1, 1, 1},   // 7
+                                                           {2, 0, 0},   // 8
+                                                           {2, 1, 0},   // 9
+                                                           {2, 0, 1},   // 10
+                                                           {2, 1, 1}}}; // 11
+
+    for (unsigned int i = 0; i < 12; ++i)
+      vertices.push_back(hexahedron[i]);
```

Comment:
I think you can do this simpler:
```
  const std::vector<Point<dim>> vertices (hexahedron.begin(), hexahedron.end());
```

Reply:
As a matter of fact, you could probably just implement things directly without the detour of the `std::array`:
```
  const std::vector<Point<dim>> vertices = {{...}, {...}};
```
If this doesn't work right away, maybe via
```
  const std::vector<Point<dim>> vertices = {Point<dim>(...), Point<dim>(...)};
```

Reply:
done